### PR TITLE
normalizes non-mapbox:// scheme when query string exists

### PR DESCRIPTION
**Pull request by jingsam**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ npm install
 Install [git](https://git-scm.com/), [node.js](https://nodejs.org/) (version 4 or greater), [GNU Make](http://www.gnu.org/software/make/), and libglew-dev
 ```bash
 sudo apt-get update &&
-sudo apt-get install build-essential git nodejs libglew-dev
+sudo apt-get install build-essential git nodejs libglew-dev libxi-dev
```
Comment (mikemorris) [inline CONTRIBUTING.md]: What's this for?

Comment (jingsam) [inline CONTRIBUTING.md]: libxi-dev is one of the dependencies. Wtihout it, `npm install` would fail

**Comment by jingsam** [comment]
@lucaswoj

**Comment by lucaswoj** [comment]
Fantastic! Thanks @jingsam.

**Comment by jingsam** [comment]
I found that https://github.com/mapbox/mapbox-gl-js/blob/master/js/util/mapbox.js#L54 would be problematic when `url` has querystring.

As I pointed out previously, url is more complicated than we think. We should use urlObject than url string concatenating. But I found there are still many url string concatenating in `mapbox.js`.

So, I would like to refactor `mapbox.js`, any suggestions?

**Comment by mikemorris** [comment]
We should take a look at the URL handling in mapbox-gl-native as well, I think there's a problem with non-mapbox:// sprite URLs, but I haven't been able to sufficiently investigate this yet.

Refs https://github.com/mapbox/mapbox-gl-native/issues/4922#issuecomment-225724657

**Pull request merged**